src/Nethermind/Nethermind.HealthChecks/NodeHealthChecks.cs

```diff
@@ -0,0 +1,68 @@
+﻿//  Copyright (c) 2020 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Nethermind.JsonRpc.Modules;
+using Nethermind.JsonRpc.Modules.Eth;
+using Nethermind.JsonRpc.Modules.Net;
+
+namespace Nethermind.HealthChecks
+{
+    public class NodeHealthCheck: IHealthCheck
+    {
+        private readonly IRpcModuleProvider _rpcModuleProvider;
+        public NodeHealthCheck(IRpcModuleProvider rpcModuleProvider)
+        {
+            _rpcModuleProvider = rpcModuleProvider ?? throw new ArgumentNullException(nameof(rpcModuleProvider));
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {            
+            IEthModule ethModule = (IEthModule) await _rpcModuleProvider.Rent("eth_syncing", false);
+            INetModule netModule = (INetModule) await _rpcModuleProvider.Rent("net_peerCount", false);
+            try
+            {
+                long netPeerCount = (long) netModule.net_peerCount().GetData();
+                SyncingResult ethSyncing = (SyncingResult) ethModule.eth_syncing().GetData();
+
+                if(ethSyncing.IsSyncing == false && netPeerCount > 0)
+                {
+                    return HealthCheckResult.Healthy(description: $"The node is now fully synced with a network, number of peers: {netPeerCount}");
+                }
+                else if (ethSyncing.IsSyncing == false && netPeerCount == 0)
+                {
+                    return HealthCheckResult.Unhealthy(description: $"The node has 0 peers connected");
+                }
+                else
```

Comment:
if you want you can delete this else statement and simply return after previous if and else if was false
just a minor tip, can be ignored

Reply:
What I started doing in all my new code is to have only one return statement per method. I was avoiding this for a long time but now I have to admit it helps a lot.